src/lib.rs

```diff
@@ -309,11 +312,32 @@ impl KVMIntrospectable for KVMi {
     fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, kvm_msrs), Error> {
         let mut regs: kvm_regs = unsafe { mem::MaybeUninit::<kvm_regs>::zeroed().assume_init() };
         let mut sregs: kvm_sregs = unsafe { mem::MaybeUninit::<kvm_sregs>::zeroed().assume_init() };
-        let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        //let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        let size = 6;
+        let layout = Layout::from_size_align(
+            mem::size_of::<u32>()
+                + mem::size_of::<u32>()
+                + size * (mem::size_of::<kvm_msr_entry>()),
+            cmp::max(mem::align_of::<u32>(), mem::align_of::<kvm_msr_entry>()),
+        )
+        .unwrap();
+        /*let layout = Layout::from_size_align(
+            size * (mem::size_of::<kvm_msr_entry>()),
+            mem::align_of::<kvm_msr_entry>(),
+        )
+        .unwrap();*/
+        let mut msr_ptr = alloc(layout) as *mut kvm_msrs;
+        let mut msrs_slice = unsafe { slice::from_raw_parts(msr_ptr, 1) };
```

Comment:
Does this even run? I meant that you should create a slice starting from the address of `msrs.entries` because that's where the array starts.